### PR TITLE
i2c: Move i2c to top folder

### PR DESCRIPTION
**Pull request by vireshk**
The workspace section in Rust documentation suggests adding new crates
in the top level directory itself instead of the src folder. Remove the
top level src folder.

Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>

### CODE REVIEW

**Comment by vireshk** [comment]
@andreeaflorescu Is this reorg okay ?

**Review by andreeaflorescu — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ crates.
 
 Here is the list of device backends that we support:
 
-- [I2C](https://github.com/rust-vmm/vhost-device/blob/master/src/i2c/README.md)
+- [I2C](https://github.com/rust-vmm/vhost-device/blob/master/i2c/README.md)
```
Comment (lauralt) [inline README.md:11]: nit: master should also be replaced with main.

**Review by lauralt — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (vireshk) [reply README.md:11]: Will fix that, thanks for noticing it. Though the link still works somehow, I think github is making it work somehow or something else.

Comment (lauralt) [reply README.md:11]: Yup, no rush. This seems to be the case: https://github.blog/changelog/2020-07-17-links-to-deleted-branches-now-redirect-to-the-default-branch/.

**Pull request merged**